Bluetooth: controller: re-add missing hci_vendor.c reference

fabiobaltieri opened the pull request:
Add hci/nordic/hci_vendor.c back to the list when building for nRF platforms, this was inadvertently removed in

7bd2804c24 Bluetooth: Controller: Build vendor LLL as library

and is currently causing a:

<wrn> bt_id: No static addresses stored in controller

and random address on nRF devices.

fabiobaltieri committed:
Bluetooth: Controller: re-add missing hci_vendor.c reference

Move the nordic and openisa vendor file under ll_sw and add them back to
the list when building for nRF platforms, this was inadvertently removed
in

7bd2804c24 Bluetooth: Controller: Build vendor LLL as library

and is currently causing a:

<wrn> bt_id: No static addresses stored in controller

and random address on nRF devices.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>
Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak commented on subsys/bluetooth/controller/CMakeLists.txt:
```diff
@@ -187,6 +187,11 @@ zephyr_library_include_directories_ifdef(
   hci/nordic
   )
 
+zephyr_library_sources_ifdef(
+  CONFIG_SOC_COMPATIBLE_NRF
+  hci/nordic/hci_vendor.c
+  )
+
```
Thank you for discovering.

1. Add it here: https://github.com/cvinayak/zephyr/blob/2de71fa1d7eb3e5b29b52dd0be9d1f6eb8b52572/subsys/bluetooth/controller/ll_sw/nordic/CMakeLists.txt#L87 and respectively for `openisa` too.

as:
```c
zephyr_library_sources(
  hci/hci_vendor.c
  )
```

2. Move the vendor specific {nordic, openisa} header and source files too into `ll_sw/{nordic,openisa}/hci/` 

(maybe easy said by me, but give a try else I can do it).

fabiobaltieri replied:
okay, had to adjust a path as well, did I get this right?

fabiobaltieri commented:
@cvinayak applied the comment but also noticed that the previous twister run failed on `nrf52_bsim` with

```
/usr/bin/ld.bfd: zephyr/subsys/bluetooth/controller/libsubsys__bluetooth__controller.a(hci_vendor.c.obj): in function `hci_vendor_read_static_addr':
/__w/zephyr/zephyr/subsys/bluetooth/controller/hci/nordic/hci_vendor.c:19: multiple definition of `hci_vendor_read_static_addr'; app/libapp.a(hci_vendor_stub.c.obj):/__w/zephyr/zephyr/tests/bluetooth/df/common/src/hci_vendor_stub.c:17: first defined here
```

Guess that file should be skipped when building for bsim, how should it be handled? unsure

fabiobaltieri commented:
Tentatively put the file behind a `if (NOT CONFIG_SOC_SERIES_BSIM_NRFXX)`

cvinayak commented:
> Tentatively put the file behind a `if (NOT CONFIG_SOC_SERIES_BSIM_NRFXX)`

Sorry my oversight, previously it was like this:
```c
zephyr_library_sources_ifdef(
  CONFIG_SOC_FAMILY_NRF
  hci/nordic/hci_vendor.c
  )
```

Maybe the below is better (and works as intended for both SoC and BSIM?):
```c
zephyr_library_sources_ifdef(
  CONFIG_SOC_FAMILY_NRF
  hci/hci_vendor.c
  )
```

@aescolar Need your review here as to how it should be these days...

cvinayak commented:
I will push the change and see...

fabiobaltieri commented:
Cool, CI is happy

aescolar commented:
> @cvinayak applied the comment but also noticed that the previous twister run failed on nrf52_bsim with

@fabiobaltieri Do you recall or can you locate which test was failing?

aescolar commented:
> @aescolar Need your review here as to how it should be these days...

By now I guess we can just get this is as is, as this issue of not building this file for simulation was always there.
I'd like to fix it, but given what Fabio said above, it may be better to follow it up on another PR

fabiobaltieri commented:
> @fabiobaltieri Do you recall or can you locate which test was failing?

None of them were failing, I noticed the addressed changing randomly while working on an out-of-tree project and reproduced with the beacon one but could have used any of those.

> By now I guess we can just get this is as is, as this issue of not building this file for simulation was always there.

Yeah this is annoying to anyone working on main, may be a good idea to get it in sooner rather than later.

aescolar committed:
Bluetooth: Controller: Build hci_vendor.c also for simulation

The FICR is also modelled in simulation, but changing this
conditional compile was forgotten, so it was only being
built for real HW.
A test had provided its own version of this function so it would
avoid a runtime warning due to the lack of static address. But
this needs to be removed as it conflicts with the function
provided now always by the nordic controller SW.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar commented:
@fabiobaltieri @cvinayak  I think this is the proper fix on top of this PR so this file is also built for simulation:
https://github.com/aescolar/zephyr/commit/93ec925fc3d4bb4dc958187dd765c5504067f8ae

aescolar reviewed (dismissed):
I'm ok with this being merged as is.
You can also take https://github.com/aescolar/zephyr/commit/93ec925fc3d4bb4dc958187dd765c5504067f8ae into this PR if you want,
or I can submit a follow up with that commit right after this one is merged.
You decide

fabiobaltieri commented:
> You can also take [aescolar@93ec925](https://github.com/aescolar/zephyr/commit/93ec925fc3d4bb4dc958187dd765c5504067f8ae) into this PR if you want,

Yup, pulled that in this PR.

cvinayak commented:
Minor nitpick on the title: `subsys/bt/ctlr/nordic: Build hci_vendor.c also for simulation`  to be `Bluetooth: Controller: Build hci_vendor.c also for simulation`

fabiobaltieri commented:
> `Bluetooth: Controller: Build hci_vendor.c also for simulation`

Fixed
